pkg/discovery/network/generic.go

```diff
@@ -78,22 +77,31 @@ func discoverNetwork(ctx context.Context, client controllerClient.Client) (*Clus
 
 func findClusterIPRange(ctx context.Context, client controllerClient.Client) (string, error) {
 	clusterIPRange, err := findClusterIPRangeFromApiserver(ctx, client)
-	if err != nil || clusterIPRange != "" {
-		return clusterIPRange, err
+	if clusterIPRange != "" {
```

Comment:
Why did you remove the error checking here and other places below? This hides the error which isn't best practice. An error here means it could not determine if the IP range could be obtained, most likely because the API server wasn't available, in which case we want to abort and report the error.

Reply:
Well, It seems I understand wrong the purpose of these error checking. I thought this function is to detect the available ClusterIP range, but it seems to detect the availability of component/service related to ClusterIP. I will cancel the related modifications later. 

In addition, I think the point 1&2 should be retained, but for point 3, I wonder if add the redundant finding from kube-controller-manager.

Reply:
I'm not sure I follow what you're saying. As I explained above, there is nothing wrong with the current error checking  - it should not be changed.

Reply:
I can understand it. So, each error should be checked and not be covered by assignments. However, there is no doubt that `err != nil || clusterIPRange != ""` condition will cover/skip the following statements. This point must be corrected. I'll modify the PR correspondingly.

Reply:
No - the error checking was fine as is. Given your recent changes and the scenario I outlined above, it would return a valid cluster IP **and** an error which is non-sensical and non-idiomatic.

Reply:
Thanks for your patience. I know what I misunderstood. `err != nil || clusterIPRange != ""` is not always true. There is a situation not finding cluster ip range from the args of apiserver/controller-manager. 

I have modified my patch, removing the changes of error checking.